frontend/pages/admin/components/DownloadFileButtons/DownloadCSR.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { FormEvent, useCallback, useMemo, useState } from "react";
+
+import mdmAppleApi from "services/entities/mdm_apple";
+
+import Icon from "components/Icon";
+import Button from "components/buttons/Button";
+import { RequestState, downloadBase64ToFile } from "./helpers";
+
+interface IDownloadCSRProps {
+  baseClass: string;
+  onSuccess?: () => void;
+  onError?: (e: unknown) => void;
+}
+
+const downloadCSRFile = (data: { csr: string }) => {
+  downloadBase64ToFile(data.csr, "fleet-mdm-apple.csr");
+};
+
+// TODO: why can't we use Content-Dispostion for these? We're only getting one file back now.
+
+const useDownloadCSR = ({
+  onSuccess,
+  onError,
+}: Omit<IDownloadCSRProps, "baseClass">) => {
+  const [downloadState, setDownloadState] = useState<RequestState>(undefined);
+
+  const handleDownload = useCallback(
+    async (evt: FormEvent) => {
+      evt.preventDefault();
+      setDownloadState("loading");
+      try {
+        const data = await mdmAppleApi.requestCSR();
+        downloadCSRFile(data);
+        setDownloadState("success");
+        onSuccess && onSuccess();
+      } catch (e) {
+        setDownloadState("error");
+        onError && onError(e);
+      }
+    },
+    [onError, onSuccess]
+  );
+
+  const memoized = useMemo(
+    () => ({
+      downloadState,
+      handleDownload,
+    }),
+    [downloadState, handleDownload]
+  );
+
+  return memoized;
+};
```

Comment:
why the hook? I see in the PR we're doing this in a couple of places but this is introducing a new pattern when we already tend to just write this within the component. I do like how it cleans up the component though. Maybe this is a pattern we can talk about in a FE sync

Also should we be using `dowloadState` in the component? it's set in this hook but not read anywhere in the component.

Reply:
I had some ambitions to make this a more generic `useDownloadFile` hook but haven't had a chance to distill things yet. I'll add it to the agenda to discuss.